File: _정답코드/3주차/04_슬라이딩_윈도우_2/solution.py
def solution(numbers, window):
    n = len(numbers)
    window_sum = 0

    for i in range(window):
        window_sum += numbers[i]

    answer = [window_sum]

    for i in range(window, n):
        window_sum += numbers[i] - numbers[i - window]
        answer.append(window_sum)

    return answer

File: _정답코드/3주차/04_슬라이딩_윈도우_2/solution.cpp
#include <vector>
using namespace std;

vector<int> solution(vector<int> numbers, int window) {
    int n = numbers.size();
    int windowSum = 0;

    for (int i = 0; i < window; i++) {
        windowSum += numbers[i];
    }

    vector<int> answer;
    answer.push_back(windowSum);

    for (int i = window; i < n; i++) {
        windowSum += numbers[i] - numbers[i - window];
        answer.push_back(windowSum);
    }

    return answer;
}

